### PR TITLE
Making ironic helm chart configMap names static

### PR DESCRIPTION
**Pull request by chajain**
Signed-off-by: Charu Jain <cjain@suse.com>

### CODE REVIEW

File: helm-charts/ironic/values.yaml [hunk 1]
```diff
@@ -264,10 +264,10 @@ ironic:
     env:
       - name: MARIADB_PASSWORD
         secretKeyRefKey: password
-        secretKeyRefName: baremetal-operator-mariadb
```
Comment (guangyee) [inline helm-charts/ironic/values.yaml:267]: Wow, I am surprised how it was working before consider the old name was `{{ include "ironic.fullname" . }}-baremetal-operator-mariadb`

File: helm-charts/ironic/templates/deployment.yaml [hunk 2]
```diff
@@ -64,7 +64,7 @@ spec:
           envFrom:
           {{- range .envFrom }}
             - configMapRef:
-                name: {{ $relname }}-{{ .configMapRefname }}
```
Comment (guangyee) [inline helm-charts/ironic/templates/deployment.yaml:67]: Not sure if you really need this change.

File: helm-charts/ironic/templates/deployment.yaml [hunk 3]
```diff
@@ -75,12 +75,12 @@ spec:
                 {{- if .secretKeyRefKey }}      
                 secretKeyRef:
                   key: {{ .secretKeyRefKey }}
-                  name: {{ $relname }}-{{ .secretKeyRefName }}
```
Comment (guangyee) [inline helm-charts/ironic/templates/deployment.yaml:78]: not sure if you need this change

File: helm-charts/ironic/templates/deployment.yaml [hunk 4]
```diff
@@ -75,12 +75,12 @@ spec:
...
+                  name: {{ .secretKeyRefName }}
                 {{- end }}
                 {{- if .configMapKeyRefKey }}
                 configMapKeyRef:
                   key: {{ .configMapKeyRefKey }}
-                  name: {{ $relname }}-{{ .configMapKeyRefName }}
```
Comment (guangyee) [inline helm-charts/ironic/templates/deployment.yaml:83]: not sure if you need this change

File: helm-charts/ironic/values.yaml (hunk 1 above)
Comment (guangyee) [reply helm-charts/ironic/values.yaml:267]: Looks like we're missing `IRONIC_USE_MARIADB: "true"` from the ConfigMap and therefore sqllite was use as default. I think we'll need to add it back so we can default to MariaDB and optionally use sqlite if needed.

**Comment by rtamalin** [comment]
Testing these changes in conjunction with #32 I see improvement; the pod starts up, but that process takes a very very very very long time, and it appears that we repeatedly pull fresh copies of the same ironic image:

```
metal@metal3:~/baremetal/helm-charts$ kubectl describe pods --namespace metal-cubed heavy-metal-metal3-ironic-5496488d54-jj9hx
...
Events:
  Type     Reason     Age                   From               Message
  ----     ------     ----                  ----               -------
  Normal   Scheduled  11m                   default-scheduler  Successfully assigned metal-cubed/heavy-metal-metal3-ironic-5496488d54-jj9hx to metal3
  Normal   Pulling    11m                   kubelet            Pulling image "quay.io/metal3-io/ironic-ipa-downloader"
  Normal   Pulled     10m                   kubelet            Successfully pulled image "quay.io/metal3-io/ironic-ipa-downloader" in 26.475016886s
  Normal   Created    10m                   kubelet            Created container ironic-ipa-downloader
  Normal   Started    10m                   kubelet            Started container ironic-ipa-downloader
  Normal   Pulling    10m                   kubelet            Pulling image "quay.io/metal3-io/ironic"
  Normal   Pulled     10m                   kubelet            Successfully pulled image "quay.io/metal3-io/ironic" in 26.320096898s
  Normal   Created    10m                   kubelet            Created container ironic-dnsmasq
  Normal   Started    10m                   kubelet            Started container ironic-dnsmasq
  Normal   Pulling    10m                   kubelet            Pulling image "quay.io/metal3-io/mariadb"
  Normal   Pulled     9m47s                 kubelet            Successfully pulled image "quay.io/metal3-io/mariadb" in 26.38166013s
  Normal   Created    9m46s                 kubelet            Created container mariadb
  Normal   Started    9m46s                 kubelet            Started container mariadb
  Normal   Pulling    9m46s                 kubelet            Pulling image "quay.io/metal3-io/ironic"
  Normal   Pulled     9m45s                 kubelet            Successfully pulled image "quay.io/metal3-io/ironic" in 694.550806ms
  Normal   Created    9m45s                 kubelet            Created container ironic
  Normal   Started    9m45s                 kubelet            Started container ironic
  Normal   Pulling    9m45s                 kubelet            Pulling image "quay.io/metal3-io/ironic"
  Normal   Pulled     9m44s                 kubelet            Successfully pulled image "quay.io/metal3-io/ironic" in 604.279152ms
  Normal   Created    9m43s                 kubelet            Created container ironic-log-watch
  Normal   Started    9m42s                 kubelet            Started container ironic-log-watch
  Normal   Pulling    9m42s                 kubelet            Pulling image "quay.io/metal3-io/ironic"
  Normal   Pulled     9m42s                 kubelet            Successfully pulled image "quay.io/metal3-io/ironic" in 639.695567ms
  Normal   Created    9m41s                 kubelet            Created container ironic-inspector
  Normal   Started    9m41s                 kubelet            Started container ironic-inspector
  Normal   Pulling    9m41s                 kubelet            Pulling image "quay.io/metal3-io/ironic"
  Warning  Unhealthy  72s (x15 over 9m12s)  kubelet            Liveness probe failed:
```

**Comment by rtamalin** [comment]
Ok so the above errors were partially user error - I was missing some system specific override settings.

**Comment by rtamalin** [comment]
So in general looks good, and I think that once I test it in a properly setup environment, with appropriate overrides it likely will start correctly

File: helm-charts/ironic/values.yaml (hunk 1 above)
Comment (chajain) [reply helm-charts/ironic/values.yaml:267]: > Wow, I am surprised how it was working before consider the old name was `{{ include "ironic.fullname" . }}-baremetal-operator-mariadb`

Before I was joining the release-name with the secretKeyRefName in the template and that was matching with the secret key being created. Now I have made all the configMap and secret-key name static

File: helm-charts/ironic/templates/deployment.yaml (hunk 2 above)
Comment (chajain) [reply helm-charts/ironic/templates/deployment.yaml:67]: Since I am using static configmapName. I no longer need to append that with release name

**Review by guangyee — APPROVED** [review]
I retested everything and able to successfully provision a cluster.

**Pull request merged**